src/EnergyPlus/InputProcessor.cc

```diff
@@ -3755,6 +3755,30 @@ namespace InputProcessor {
 
 	}
 
+	std::string
+	deAllCaps( std::string const & inString )
+	{
+		// Jason Glazer - August 2015
+		// turn all caps into caps only when first letter in word
+		std::string returnString;
+		bool wasSpace = true; //start out assuming a space so first character is in caps
+		for ( std::string::size_type i = 0; i != inString.size(); i++ ){
+			char curChar = inString[i];
+			if ( wasSpace ){
+				returnString.push_back( inString[i] );
+			}
+			else {
+				if ( curChar > 64 && curChar < 91 ){
+					curChar = curChar + 32;
+				}
+				returnString.push_back( curChar );
+			}
+			wasSpace = ( curChar == 32 );
+		}
+		return returnString;
+	}
+
+
```

Comment:
If the original names are not available then solutions that are probably faster:
* stream manipulator (you probably don't want to learn how)
* Convert the characters in place if the passed in string is not needed in its uppercase form anymore
* Construct the new string as a copy of the old so you are sure only one allocation occurs then convert the first characters in each word in place (or do a reserve before your push_back loop)

This is normally called title case so that might be better in the function name and comments than deAllCaps.

The assumption that all the string operations weren't called enough to be a performance issue was inaccurate and led to a lot of work after the C++ conversion so I encourage finding a worst-case performance test to verify such an assumption if a potentially slow solution is really needed.